modules/appeals_api/app/controllers/appeals_api/notice_of_disagreements/v0/notice_of_disagreements/evidence_submissions_controller.rb

```diff
@@ -19,7 +20,14 @@ class EvidenceSubmissionRequestValidatorError < StandardError; end
 
     def show
       submission = AppealsApi::EvidenceSubmission.find_by(guid: params[:id])
-      raise Common::Exceptions::RecordNotFound, params[:id] unless submission
+
```

Comment:
Since I'm now looking up the appeal itself to check that its ICN matches the ICN in the token (and raising this same error if the appeal is not found), I customized the "not found" error messages to specify the type of record that was not found